ClassPaulita/GuessTheCardGame.cs

```diff
@@ -1,107 +1,174 @@
     {
-        return userChoice == 'p' || userChoice == 'P';
-    }
+        private string spacing = "\n\t\t\t\t\t\t";
 
-    private bool UserchoseToSeeInstructions(char userChoice)
-    {
-        return userChoice == 'i' || userChoice == 'I';
-    }
-
-    private bool UserChosetoSeeBoard(char userChoice)
-    {
-        return userChoice == 'b' || userChoice == 'B';
-    }
-
-    private bool UserChoseToQuit(char userChoice)
-    {
-        return userChoice == 'q' || userChoice == 'Q';
-    }
+        private bool UserChoseToPlay(char userChoice)
+        {
+            return userChoice == 'p' || userChoice == 'P';
+        }
 
-    private bool IsValidInput(string input)
-    {
-        if (input == null || input.Length < 1)
-            return false;
+        private bool UserchoseToSeeInstructions(char userChoice)
+        {
+            return userChoice == 'i' || userChoice == 'I';
+        }
 
-        char userChoice = input[0];
+        private bool UserChosetoSeeBoard(char userChoice)
+        {
+            return userChoice == 'b' || userChoice == 'B';
+        }
 
-        if (UserChoseToPlay(userChoice) || UserchoseToSeeInstructions(userChoice) || UserChosetoSeeBoard(userChoice) || UserChoseToQuit(userChoice))
+        private bool UserChoseToQuit(char userChoice)
         {
-            return true;
+            return userChoice == 'q' || userChoice == 'Q';
         }
-       
-        return false;
-    }
 
-    private string PrintMenu()
-    {
-        string spacing = "\n\t\t\t\t\t\t";
+        private bool IsUserChoiceValid(string input)
+        {
+            if (input == null || input.Length < 1)
+                return false;
 
-        string menu = $"{spacing}Press P To Play" +
-                      $"{spacing}Press I For Instructions" +
-                      $"{spacing}Press B To See The High Score Board" +
-                      $"{spacing}Press Q To Quit";
+            char userChoice = input[0];
 
-        Console.WriteLine(menu);
+            if (UserChoseToPlay(userChoice) || UserchoseToSeeInstructions(userChoice) || UserChosetoSeeBoard(userChoice) || UserChoseToQuit(userChoice))
+            {
+                return true;
+            }
 
-        string userChoice = Console.ReadLine();
+            return false;
+        }
 
-        while (!IsValidInput(userChoice))
+        private string PrintMenu()
         {
-            Console.WriteLine("\n\tOoooops! You pressed an invalid option. Please try again.");
+
+            string menu = $"{spacing}Press P To Play" +
+                          $"{spacing}Press I For Instructions" +
+                          $"{spacing}Press B To See The High Score Board" +
+                          $"{spacing}Press Q To Quit";
 
             Console.WriteLine(menu);
 
-            userChoice = Console.ReadLine();
-        }
+            string userChoice = Console.ReadLine();
 
-        return userChoice;
-    }
+            while (!IsUserChoiceValid(userChoice))
```

Comment:
If I were just looking at function names I would not know the difference between IsUserChoiceValid and IsUserInputValid.  Please rename them so they are clearer.

Reply:
True. I changed the names to IsUserGuessValid() and IsUserMenuChoiceValid(). I know they are a bit long but us the only way I could think of to make them more distinctive.